src/sync/chase_lev.rs

```diff
@@ -184,46 +184,43 @@ impl<T> Deque<T> {
             b = self.bottom.load(Relaxed);
         }
         a.put(b, data);
-        fence(Release);
-        self.bottom.store(b + 1, Relaxed);
+        self.bottom.store(b + 1, Release);
```

Comment:
Yes, @alexcrichton is right -- I silently sneaked in something without explicitly saying it.  It probably should be done in a separate PR.

Having said that, I think release store also does its job: the release fence's sole purpose was releasing the view (think: information) gathered so far to the write of `self.bottom`.  The same job can be done via a release write to `self.bottom`.

In fact, I think several orderings in this implementation of Chase-Lev deque is stronger than what is necessary.  I will make another PR on the orderings.  In the mean time, I will roll back this change.